KsApi/ServiceTests.swift

```diff
@@ -120,55 +120,114 @@ final class ServiceTests: XCTestCase {
     XCTAssertNil(result)
   }
 
-  func test_Array_SwiftDecodeModel_ValidModel() {
+  func testSwiftDecodeModel_ValidModel_Combine() {
+    let expectedResult = MySwiftModel(
+      array: ["string1", "string2"],
+      bool: true,
+      dict: ["key1": "value1", "key2": "value2"],
+      id: 5,
+      name: "Swift Name"
+    )
+
     let jsonData: String = """
-    [
-       {
-          "array":[
-             "string1",
-             "string2"
-          ],
-          "bool":true,
-          "dict":{
-             "key1":"value1",
-             "key2":"value2"
-          },
-          "id":5,
-          "name":"Swift Name"
-       },
-       {
-          "array":[
-             "string1",
-             "string2"
-          ],
-          "bool":true,
-          "dict":{
-             "key1":"value1",
-             "key2":"value2"
-          },
-          "id":6,
-          "name":"Swift Name"
-       }
-    ]
+    {
+        "array": ["string1", "string2"],
+        "bool": true,
+        "dict": {"key1": "value1", "key2": "value2"},
+        "id": 5,
+        "name": "Swift Name"
+    }
     """
 
-    let expectedResult = try! JSONDecoder().decode([MySwiftModel].self, from: jsonData.data(using: .utf8)!)
-
-    let model: SignalProducer<[MySwiftModel], ErrorEnvelope> = Service()
-      .decodeModels(jsonData.data(using: .utf8)!)
-    let result = try! model.single()?.get()
-    XCTAssertEqual(result, expectedResult)
+    let data = jsonData.data(using: .utf8)!
+    let result: Result<MySwiftModel?, ErrorEnvelope> = Service()
+      .decodeModel(data: data, ofType: MySwiftModel.self)
+
+    switch result {
+    case let .success(decodedModel):
+      if let decodedModelUnwrapped = decodedModel {
+        XCTAssertEqual(expectedResult, decodedModelUnwrapped)
+      } else {
+        XCTFail("Decoded model is nil")
+      }
+    case .failure: XCTFail("Failed to decode model")
+    }
   }
 
-  func test_Array_SwiftDecodeModel_EmptyArray() {
+  func testSwiftDecodeModel_InvalidModel_Combine() {
     let jsonData: String = """
-    [
-    ]
+    {
+        "notAValidKey": "foo"
+    }
     """
 
-    let model: SignalProducer<[MySwiftModel], ErrorEnvelope> = Service()
-      .decodeModels(jsonData.data(using: .utf8)!)
-    let result = try! model.single()?.get()
-    XCTAssertEqual(result?.count, 0)
+    let data = jsonData.data(using: .utf8)!
+    let result: Result<MySwiftModel?, ErrorEnvelope> = Service()
+      .decodeModel(data: data, ofType: MySwiftModel.self)
+
+    switch result {
+    case .success: XCTFail("Decode should have failed.")
+    case let .failure(error): XCTAssertNotNil(error)
+    }
   }
+
+  /*
+   These two tests shouldn't actually *run* in our testing environment, because we don't want to make real network requests
+   in our real tests. But it's a useful little utility for comparing the Combine and ReactiveSwift version of the
+   API V1 call Request.
+   */
+
+  /*
```

Comment:
I commented these tests out, but ran them locally to ensure that they passed. We don't have a way to mock HTTP at this low of a level - hence these tests that actually _make a real network request_. Obviously something we don't want to do on CircleCI, since it would be a flaky mess.

We could consider adding something like `OHHTTPStubs` for low-level HTTP mocking, or one of those `NSURLProtocol` implementations that are floating around. But this just felt out of scope for this PR. Hence, my compromise option - tests you can turn on yourself if you need them 😉